server/container/testing/src/test/java/org/apache/james/deployment/DeploymentTest.java

```diff
@@ -0,0 +1,190 @@
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+public class DeploymentTest {
+
+    private static final String LOCALHOST = "localhost";
+    private static final String SIMPSON = "simpson";
+    private static final String HOMER = "homer@" + SIMPSON;
+    private static final String HOMER_PASSWORD = "secret";
+    private static final String BART = "bart@" + SIMPSON;
+    private static final String BART_PASSWORD = "tellnobody";
+    private static final int JMAP_PORT = 80;
+    private static final int SMTP_PORT = 25;
+    private static final int IMAP_PORT = 143;
+    private static final int WEBADMIN_PORT = 8000;
+
+    private static final Logger logger = LoggerFactory.getLogger(DeploymentTest.class);
+
+    public Network network = Network.newNetwork();
+
+    public GenericContainer<?> cassandra = new CassandraContainer()
+        .withNetwork(network)
+        .withNetworkAliases("cassandra");
+    
+    public GenericContainer<?> elasticsearch = new GenericContainer<>("elasticsearch:2.4.6")
+        .withNetwork(network)
+        .withNetworkAliases("elasticsearch");
+
+    public GenericContainer<?> james = new GenericContainer<>("linagora/james-project:latest")
+        .withExposedPorts(JMAP_PORT, SMTP_PORT, IMAP_PORT, WEBADMIN_PORT)
+        .withCopyFileToContainer(MountableFile.forClasspathResource("/webadmin.properties"), "/root/conf/")
+        .withCopyFileToContainer(MountableFile.forClasspathResource("/jmap.properties"), "/root/conf/")
+        .withCopyFileToContainer(MountableFile.forClasspathResource("/keystore"), "/root/conf/")
+        .withCopyFileToContainer(MountableFile.forClasspathResource("/jwt_publickey"), "/root/conf/")
+        .withNetwork(network)
+        .withLogConsumer(log -> logger.info(log.getUtf8String()))
+        .waitingFor(new HttpWaitStrategy().forPort(WEBADMIN_PORT).forStatusCodeMatching(status -> true));
```

Comment:
How do you know webadmin is the last started thing?

IMO we would need a probe that gets its data updated when James is started, and webadmin seems a good idea to expose that probe

Reply:
we should start implementing a health-check endpoint

Reply:
That's what I started doing...